### ISSUE TITLE
Call for use cases and examples

### ISSUE THREAD

**Issue by rlouf** [rlouf/mcx#3]
ˋmcx` is only interesting if it can be used and we need more example to showcase the way the library can be used. Examples are a great way to get acquainted with the library and make a first contribution.

We are particularly looking for examples that use:

- Batching (vectorized sampling) on CPU, GPU or TPU;
- Sequential inference;
- Bayesian Neural Network.

**Comment by rlouf** [comment]
- [ ] Eight schools
- [ ] Baseball
- [ ] Example of dynamic stopping rule (Rhat, ESS)
- [ ] SIR model: ode or not, gaussian process prior on R0
- [ ] A/B testing with sequential data input
- [ ] Gaussian mixture
- [ ] Bayesian workflow

**Comment by miretchin** [comment]
Statistical rethinking examples would be really great to have! Some of the chapters won't yet be possible because it looks like the required distributions are missing, particularly the discrete distributions Beta Binomial, Gamma Poisson, Zero-Inflated Poisson (or zero-inflated negative binomial), and Ordered Logit / Probit.

**Comment by rlouf** [comment]
Indeed! I will bump the implementation of discrete distributions so that the first release contains enough functionalities to implement most of the examples from Statistical Rethinking.

**Comment by elanmart** [comment]
Hi @rlouf , if I wanted to start implementing the Statistical Rethinking examples in `mcx`, should I go with `master` branch? Or the `compiler-refactor`?

**Comment by rlouf** [comment]
Awesome! `compiler-refactor` is the branch with the most recent version of the API. By the time you finish it should be merged :) 

I started here but there's almost nothing: https://github.com/mcx-ppl/statistical-rethinking-mcx. Note that the `sample` method of distributions returns a 1-dimension array for now, but will soon be a float (PR on the way).

**Comment by elanmart** [comment]
@rlouf thanks, I'm actually using the book as intro to bayesian methods, so I was planning to implement stuff myself anyway :) 

I started with chapters 2 and 3 today: https://github.com/elanmart/rethinking-2

**Comment by rlouf** [comment]
Nice! Let me know if you bump into an issue with MCX, something not working or missing.
